diff --git a/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/Inc/stm32f4xx_hal_conf.h b/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/Inc/stm32f4xx_hal_conf.h
index 4ae6ff5b1..b519eaa9f 100644
--- a/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/Inc/stm32f4xx_hal_conf.h
+++ b/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/Inc/stm32f4xx_hal_conf.h
@@ -56,7 +56,7 @@
 /* #define HAL_LTDC_MODULE_ENABLED     */
 #define HAL_PWR_MODULE_ENABLED
 #define HAL_RCC_MODULE_ENABLED      
-/* #define HAL_RNG_MODULE_ENABLED       */   
+#define HAL_RNG_MODULE_ENABLED
 /* #define HAL_RTC_MODULE_ENABLED       */
 /* #define HAL_SAI_MODULE_ENABLED        */   
 /* #define HAL_SD_MODULE_ENABLED         */
diff --git a/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/STM32CubeIDE/Example/User/syscalls.c b/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/STM32CubeIDE/Example/User/syscalls.c
index d190edf31..94c06fe01 100644
--- a/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/STM32CubeIDE/Example/User/syscalls.c
+++ b/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/STM32CubeIDE/Example/User/syscalls.c
@@ -96,14 +96,14 @@ int _close(int file)
 }
 
 
-int _fstat(int file, struct stat *st)
+__attribute__((used)) int _fstat(int file, struct stat *st)
 {
   (void)file;
   st->st_mode = S_IFCHR;
   return 0;
 }
 
-int _isatty(int file)
+__attribute__((used)) int _isatty(int file)
 {
   (void)file;
   return 1;
diff --git a/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/Src/stm32f4xx_hal_msp.c b/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/Src/stm32f4xx_hal_msp.c
index edafa3259..377a6f9e6 100644
--- a/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/Src/stm32f4xx_hal_msp.c
+++ b/Projects/STM32F4-Discovery/Examples/UART/UART_TwoBoards_ComPolling/Src/stm32f4xx_hal_msp.c
@@ -38,6 +38,51 @@
   * @{
   */
 
+/**
+* @brief RNG MSP Initialization
+* This function configures the hardware resources used in this example
+* @param hrng: RNG handle pointer
+* @retval None
+*/
+void HAL_RNG_MspInit(RNG_HandleTypeDef* hrng)
+{
+  if(hrng->Instance==RNG)
+  {
+  /* USER CODE BEGIN RNG_MspInit 0 */
+
+  /* USER CODE END RNG_MspInit 0 */
+    /* Peripheral clock enable */
+    __HAL_RCC_RNG_CLK_ENABLE();
+  /* USER CODE BEGIN RNG_MspInit 1 */
+
+  /* USER CODE END RNG_MspInit 1 */
+
+  }
+
+}
+
+/**
+* @brief RNG MSP De-Initialization
+* This function freeze the hardware resources used in this example
+* @param hrng: RNG handle pointer
+* @retval None
+*/
+void HAL_RNG_MspDeInit(RNG_HandleTypeDef* hrng)
+{
+  if(hrng->Instance==RNG)
+  {
+  /* USER CODE BEGIN RNG_MspDeInit 0 */
+
+  /* USER CODE END RNG_MspDeInit 0 */
+    /* Peripheral clock disable */
+    __HAL_RCC_RNG_CLK_DISABLE();
+  /* USER CODE BEGIN RNG_MspDeInit 1 */
+
+  /* USER CODE END RNG_MspDeInit 1 */
+  }
+
+}
+
 /**
   * @brief UART MSP Initialization 
   *        This function configures the hardware resources used in this example: 
@@ -46,7 +91,7 @@
   * @param huart: UART handle pointer
   * @retval None
   */
-void HAL_UART_MspInit(UART_HandleTypeDef *huart)
+void HAL_UART_MspInit(UART_HandleTypeDef*)
 {  
   GPIO_InitTypeDef  GPIO_InitStruct;
   
@@ -82,7 +127,7 @@ void HAL_UART_MspInit(UART_HandleTypeDef *huart)
   * @param huart: UART handle pointer
   * @retval None
   */
-void HAL_UART_MspDeInit(UART_HandleTypeDef *huart)
+void HAL_UART_MspDeInit(UART_HandleTypeDef*)
 {
   /*##-1- Reset peripherals ##################################################*/
   USARTx_FORCE_RESET();
