diff --git a/src/drivers/sifive_fe310-g000_pll.c b/src/drivers/sifive_fe310-g000_pll.c
index 5f26d32..5f7ce9d 100644
--- a/src/drivers/sifive_fe310-g000_pll.c
+++ b/src/drivers/sifive_fe310-g000_pll.c
@@ -197,9 +197,9 @@ long __metal_driver_sifive_fe310_g000_pll_get_rate_hz(
     struct metal_clock *pllsel0 =
         __metal_driver_sifive_fe310_g000_pll_pllsel0(clock);
     long config_offset =
-        __metal_driver_sifive_fe310_g000_pll_config_offset(clock);
+        __metal_driver_sifive_fe310_g000_pll_config_offset();
     struct __metal_driver_sifive_fe310_g000_prci *config_base =
-        __metal_driver_sifive_fe310_g000_pll_config_base(clock);
+        __metal_driver_sifive_fe310_g000_pll_config_base();
     long divider_offset =
         __metal_driver_sifive_fe310_g000_pll_divider_offset(clock);
     struct __metal_driver_sifive_fe310_g000_prci *divider_base =
@@ -348,7 +348,7 @@ long __metal_driver_sifive_fe310_g000_pll_set_rate_hz(struct metal_clock *clock,
     struct metal_clock *pllsel0 =
         __metal_driver_sifive_fe310_g000_pll_pllsel0(clock);
     long config_offset =
-        __metal_driver_sifive_fe310_g000_pll_config_offset(clock);
+        __metal_driver_sifive_fe310_g000_pll_config_offset();
     long divider_offset =
         __metal_driver_sifive_fe310_g000_pll_divider_offset(clock);
     long base = __metal_driver_sifive_fe310_g000_prci_base();
